### PR TITLE
SAK-33888 - Submitted/Ungraded submission counts in the assignments list are not correct for non-electronic assignments

### PR DESCRIPTION
**Pull request by mpellicer**
Submitted/Ungraded submission counts in the assignments list are not correct for non-electronic assignments

### CODE REVIEW

**Comment by mpellicer** [comment]
@ern @bgarciaentornos Could you take a look?

I'm NOT specially happy with this implementation because the legacy logic is in the VM, I suggest to move the logic to the service, what do you think?

I've tested this in Master, with a converted 11.x instance and the results are fine now.

**Comment by ern** [comment]
Hi @mpellicer I agree with your comment would be better if there was no logic in UI. I would suggest moving the logic to the service.

**Comment by mpellicer** [comment]
Hi @ern, I moved the logic from the VM to the service.

I've decided to create two new methods in the repository in order to not put logic in there, then you have different methods for electronic and non-electronic submissions. I could add a parameter like the previous implementation but I think this is cleaner.

**Comment by ern** [comment]
So creating these new methods results in over populating the api and more code. It would be more ideal to create a single method with params maybe something like:
```
long countSubmittedSubmissionsForAssignment(String assignmentId, boolean graded, SubmissionType type);
```

**Comment by mpellicer** [comment]
I've rebased the changes, now it uses only one method with parameters.

File: assignment/api/src/java/org/sakaiproject/assignment/api/persistence/AssignmentRepository.java [hunk 1]
```diff
@@ -66,9 +66,9 @@ void newSubmission(Assignment assignment,
 
     void initializeAssignment(Assignment assignment);
 
-    long countSubmittedSubmissionsForAssignment(String assignmentId);
+    long countSubmittedSubmissionsForAssignment(String assignmentId, boolean includeUserSubmission, boolean includeSubmissionDate);
```
Comment (ern) [inline assignment/api/src/java/org/sakaiproject/assignment/api/persistence/AssignmentRepository.java]: these 2 methods should be collapsed into 1

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 2]
```diff
@@ -1497,7 +1497,11 @@ public String getSubmissionStatus(String submissionId) {
     public int getSubmittedSubmissionsCount(String assignmentReference) {
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: getSubmittedSubmissionsCount and getUngradedSubmissionsCount could be collapsed into a single method.

Comment (mpellicer) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: @ern 

In the service? Really? I think it's better to have only one method in the persistence and have two methods in the service, if not I'll need to put again logic in the UI to ask for the ungraded/submitted value to the service.

What do you think?

Comment (mpellicer) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: I've collapsed it in one method controlled by a boolean from the UI.

**Comment by ern** [comment]
Refactored this a bit as the method AssignmentRepository.countAssignmentSubmissions contained to much taint from the service. Now this method stands on its own and the service uses the method by supplying the correct params to get the result it desires.

**Pull request merged**